typing/printtyp.ml

```diff
@@ -1119,6 +1119,11 @@ and type_sch ppf ty = typexp true ppf ty
 
 and type_scheme ppf ty = reset_and_mark_loops ty; typexp true ppf ty
 
+let type_path ppf p =
+  let (p', s) = best_type_path p in
+  if (s = Id) then path ppf p'
+  else path ppf p
```

Comment:
It would be slightly more precise to call 
```ocaml
let t = tree_of_path Type p in !Oprint.out_ident ppf t
```
rather than path to also hook on the name collision detection mechanism.